src/Exp.js

```diff
@@ -24,11 +25,43 @@ module.exports = function Exp(formula) {
                 throw Error('#VALUE!');
             }
             formula.cell.v = self.calc();
-            if (typeof(formula.cell.v) === 'string') {
-                formula.cell.t = 's';
-            }
-            else if (typeof(formula.cell.v) === 'number') {
-                formula.cell.t = 'n';
+            formula.cell.t = getCellType(formula.cell.v);
+
+            if (Array.isArray(formula.cell.v && formula.cell.name && formula.cell.f && formula.cell.f.match(new RegExp(dynamicArrayFormulas.join('|'), 'i')))) {
```

Comment:
@fabiooshiro just realised this condition has a syntax error. The `Array.isArray(formula.cell.v)` is not wrapped in a bracket. So, the condition should be: 
```
(Array.isArray(formula.cell.v) && formula.cell.name && formula.cell.f && formula.cell.f.match(new RegExp(dynamicArrayFormulas.join('|'), 'i')))
```